### PR TITLE
Enable warmup also for full prompt length case in text generation

### PR DESCRIPTION
**Pull request by yeonsily**
# What does this PR do?
The prompt_length condition check for warm up is a legacy code for some previous demonstration.
We tried to show that even we do not warmup with exact (dynamic) length we still can get some performance.
But this demo case is no longer needed and all this kind of things are handled in vllm now.

Fixes # (issue)

This is a fix for performance degradation of dynamic shape run when datasets given.

For example, 
PT_HPU_LAZY_MODE=0  python3  run_generation.py  --num_beams 1 --model_name_or_path meta-llama/Llama-2-7b-chat-hf  --attn_softmax_bf16  --use_kv_cache --max_new_tokens 128 --bf16 --batch_size 8     --trim_logits --max_input_tokens -1 --n_iterations 500 --warmup 1 --dataset_name tatsu-lab/alpaca   --run_partial_dataset   --torch_compile 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by jiminha — APPROVED** [review]
LGTM!

**Review by regisss — APPROVED** [review]
